src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -1,107 +1,174 @@
 package cognite.spark.v1
 
-import cognite.spark.v1.CdpConnector.ioRuntime
+import cats.effect.{IO, Resource}
+import com.cognite.sdk.scala.v1.{File, FileCreate, Label, LabelCreate}
 import io.scalaland.chimney.dsl._
-import org.apache.spark.sql.Row
+import org.apache.spark.sql.{DataFrame, DataFrameReader, Row}
 import org.apache.spark.sql.functions.col
 import org.scalatest.{FlatSpec, Matchers, ParallelTestExecution}
 
-import scala.util.control.NonFatal
-
 class FilesRelationTest extends FlatSpec with Matchers with ParallelTestExecution with SparkTest {
-  val sourceDf = dataFrameReaderUsingOidc
-    .option("type", "files")
-    .load()
+  val testSource = s"FilesRelationTest-${shortRandomString()}"
+
+  private def makeDestinationDf(): Resource[IO, (DataFrame, String, String)] = {
+    val targetView = s"destinationFile_${shortRandomString()}"
+    val metricsPrefix = s"test.${shortRandomString()}"
+    Resource.make {
+      IO.blocking {
+        val destinationDf: DataFrame = spark.read
+          .format(DefaultSource.sparkFormatString)
+          .useOIDCWrite
+          .option("type", "files")
+          .option("collectMetrics", "true")
+          .option("metricsPrefix", metricsPrefix)
```

Comment:
Is this need for more tests than the `support updates using id and externalId` one?

Reply:
metricsPrefix part seems to be only used there, shouldn't be getting in the way for other tests I think.
EventsRelationTest has similar thing and there it is used more often so transferred in same form just in case